components/utils.py

```diff
@@ -1149,3 +1166,62 @@ def _set_upgrade_data(**kwargs):
 def _get_upgrade_data():
     with open(UPGRADE_METADATA_FILE) as f:
         return json.load(f)
+
+
+def upgrade_validation_directories(service_name):
+    try:
+        ctx_factory.get_install_properties(service_name)
+    except IOError:
+        ctx.abort_operation('Service {0} has no properties file'.format(
+            service_name))
+
+    if os.path.exists(
+            ctx_factory._get_rollback_properties_dir(service_name)):
```

Comment:
since you're using the _get_rollback_properties_dir outside the factory, rename method to be publi